docarray/base_doc/doc.py

```diff
@@ -78,35 +94,144 @@ class MyDoc(BaseDoc):
         example=os.urandom(16).hex(),
     )
 
-    class Config:
-        json_loads = orjson.loads
-        json_dumps = orjson_dumps_and_decode
-        # `DocArrayResponse` is able to handle tensors by itself.
-        # Therefore, we stop FastAPI from doing any transformations
-        # on tensors by setting an identity function as a custom encoder.
-        json_encoders = {AbstractTensor: lambda x: x}
+    if is_pydantic_v2:
+
+        class Config:
+            validate_assignment = True
+            _load_extra_fields_from_protobuf = False
+            json_encoders = {AbstractTensor: lambda x: x}
+
+    else:
+
+        class Config:
+            json_loads = orjson.loads
+            json_dumps = orjson_dumps_and_decode
+            # `DocArrayResponse` is able to handle tensors by itself.
+            # Therefore, we stop FastAPI from doing any transformations
+            # on tensors by setting an identity function as a custom encoder.
+            json_encoders = {AbstractTensor: lambda x: x}
+
+            validate_assignment = True
+            _load_extra_fields_from_protobuf = False
+
+    if is_pydantic_v2:
```

Comment:
Could you put a high level comment here explaining why v2 needs different treatment from v1?